src/effection/node.ts

```diff
@@ -0,0 +1,24 @@
+import { main as effectionMain, Context, Operation } from 'effection';
+
+export function main(operation: Operation): Context {
+  return effectionMain(({ context }) => {
+    let mainContext = context as any; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+    mainContext.spawn(function*() {
+      try {
+        yield operation;
+      } catch(e) {
+        console.error(e);
+        throw(e);
```

Comment:
Will re-throwing result in the error being logged to the console twice?

Reply:
It does not log it twice in any of my experiments. I think that Effection's main does not print errors?

Reply:
Maybe it should?

Reply:
Yeah, I definitely think there is something fishy going on. I discovered a bit about it here:

The difference is that in old effection, if the any child of the generator failed, it would throw the child's error result into the generator block and give it an opportunity to catch it from the point where it is currently yielded.

Now however, it immediately fails the  context, which causes the generator to return immediately invoking any `finally` blocks, but not providing the opportunity to actually catch the error. I think this is probably wrong. I wrote up a partial description of the issue here https://github.com/thefrontside/effection.js/issues/69

However, that question was limited to "should you be able to catch an exception that happened in a fork outside your try block." I think it's pretty clear that you should be able to catch an exception when the operation to which you're currently yielding fails and as such that this is a critical bug.

Reply:
Either way it shakes out, should we call `process.exit(-1)` instead of re-throwing the error? I kinda hate to do that on thinking of it. What if we said that the primary API for interacting with `main` is via the `Promise` API, and that if someone adds their own `catch` handlers we disable the default behavior which is `e => { console.error(e); process.exit(-1); }`

```ts
import { main } from '@effection/node';
import { app } from '../src/app';

main(app);
```

The above code prints an error to the console and exits with (-1), where as this would have some custom exit codes based on the result:

```ts
import { main } from '@effection/node';
import { app } from '../src/app';

main(app)
  .catch(e => {
    if (e.isArgumentError) {
      console.error(`There was a problem with your arguments: ${e.argumentHelpString()}`);
      process.exit(100)(
    }
    if (e.isNetworkError) {
      console.error('Could not connect servers, is your VPN enabled?');
      process.exit(200)
    }
    process.exit(-1);
  })
```

Even though `catch` blocks need fixing, I think if we have the `console.error` there, then that behavior will be difficult to override/extend. Thoughts?

Reply:
To summarize our conversation over the phone: 

We'll leave this pretty much as-is, the only difference is that instead of re-throwing, we'll `process.exit(-1)` In order to do custom error handling, you _must not_ allow your error to reach `main`